File: maow.py
from twisted.internet.protocol import Protocol, Factory
from twisted.internet.endpoints import TCP4ServerEndpoint
from twisted.internet import reactor

class MaowProtocol(Protocol):
    
    loggedIn = False

    def connectionMade(self):
        self.transport.write('Maow Server\r\nlogin: ')

    def dataReceived(self, data):
        if not self.loggedIn:
            self.username = data.strip()
            self.loggedIn = True
            self.transport.write('%s@maow> ' % (self.username,))
        else:
            cmd = data.strip()
            if len(cmd) > 0:
                if cmd == "quit":
                    self.transport.loseConnection()
                    return
            self.transport.write('mewmewmew\r\n%s@maow> ' % (self.username,))

factory = Factory()
factory.protocol = MaowProtocol
endpoint = TCP4ServerEndpoint(reactor, 2323)
endpoint.listen(factory)
reactor.run()

File: maow.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <readline/readline.h>

char *username;

void maow(){
	printf("maow!\r\n");
}

void clear(){
	printf("\033[2J\033[H");
}

int main(int argc, char **argv){
	int running;
	char* cmd;

	running = 1;
	while(running){
		cmd = readline("maow> ");
		if(cmd == (char*)NULL || strcmp(cmd, "quit") == 0){
			running = 0;
			continue;
		}
		if(strcmp(cmd, "maow") == 0){
			maow();
			continue;
		}
		if(strcmp(cmd, "clear") == 0){
			clear();
			continue;
		}
	}
	return 0;
}

